### PR TITLE
Load the default relation editor if the relation editor type setting is not set

### PR DESCRIPTION
**Pull request by suricactus**
The "widget type" from the screenshot above is automatically set to the first relation editor widget in the list, if the type was not explicitly selected before. This this is quite confusing. Now it shows the correct editor, even if the setting is missing.

image (3)

### CODE REVIEW

**Comment by nirvn** [comment]
@suricactus , side note: I think it's a problem / regression in QField too.

File: src/gui/qgsrelationwidgetregistry.h [hunk 1]
```diff
@@ -79,9 +79,25 @@ class GUI_EXPORT QgsRelationWidgetRegistry
...
+    /**
+     * Sets the default editor widget type. Does nothing if the provided widget type is not present.
+     * \param defaultWidgetType The widget type to be used by default.
+     * \since QGIS 3.22
+     */
+    void setDefaultWidgetType( const QString &widgetType );
+
+    /**
+     * Returns the default editor widget type.
+     * \since QGIS 3.22
```
Comment (3nids) [inline src/gui/qgsrelationwidgetregistry.h:92]: this is a bugfix, that should go in 3.20.

**Review by 3nids — APPROVED** [review]

File: src/gui/qgsrelationwidgetregistry.h (hunk 1 above)
Comment (suricactus) [reply src/gui/qgsrelationwidgetregistry.h]: done

**Comment by suricactus** [comment]
@nirvn I think if there is similar error, this is not related to this issue. I am working on the RelationEditor on QField these days, so I will probably spot the error.

**Pull request merged**